plugins/mongodb/mongodbembed.hpp

```diff
@@ -353,39 +384,67 @@ namespace mongodbembed
          * @param instance The instance object that is to be kept alive for multiple
          * threads to have access to.
-         * @param pool The pool object that has the information about which database
-         * and collection to connect to.
          */
-        void configure(std::unique_ptr<mongocxx::instance> && instance, std::unique_ptr<mongocxx::pool> && pool) 
+        void configure(std::unique_ptr<mongocxx::instance> && instance) 
         {
             _instance = std::move(instance);
-            _pool = std::move(pool);
         }
 
-        using connection = mongocxx::pool::entry;
+        /**
+         * @brief Creates a client object using the specified connections string. The client is
+         * added to the map of active connections.
+         * 
+         * @param connectionString The connection string for constructing the client object.
+         * 
+         * @param queryString A const char * holding the query string for hashing.
+         */
+        void create_connection(const char *connectionString, const char * queryString) 
+        {
+            // Use a hash of the connection string as the key to finding
+            // any connection objects
+            hash64_t key = rtlHash64VStr(queryString, rtlHash64VStr(connectionString, 0));
+
+            std::shared_ptr<mongocxx::client> client_ptr = std::make_shared<mongocxx::client>(mongocxx::client{mongocxx::uri{connectionString}}); 
+
+            {
+                CriticalBlock block(connLock);
+
+                clientConnections[key] = client_ptr;
+            }
+        }
 
         /**
-         * @brief Acquires a client from the pool. The calling thread will block until a
-         * connection is available.
+         * @brief Acquires a mongocxx client from the connections map.
+         * 
+         * @param connectionString A const char * holding the connection parameters.
          * 
-         * @return connection returns mongocxx::pool::entry
+         * @param queryString A const char * holding the query string for hashing.
+         * 
+         * @return A shared pointer to the mongocxx:client object for connecting to the database.
          */
-        connection get_connection() 
+        std::shared_ptr<mongocxx::client> get_connection(const char *connectionString, const char * queryString)
         {
-            return _pool->acquire();
+            // Get key for client object 
+            hash64_t key = rtlHash64VStr(queryString, rtlHash64VStr(connectionString, 0));
+
+            {
+                CriticalBlock block(connLock);
+
+                return clientConnections[key];
+            }
         }
 
     private:
         MongoDBConnection() = default;
 
-        std::unique_ptr<mongocxx::instance> _instance = nullptr;
-        std::unique_ptr<mongocxx::pool> _pool = nullptr;
+        std::unique_ptr<mongocxx::instance> _instance = nullptr;        //!< Unique pointer to the single mongocxx::instance for the program.
```

Comment:
There is an underscore because in MongoDBConnection::configure the parameter to the function is also an instance and to avoid conflicts I chose to prefix an underscore to the private member. This can be changed to something more helpful, but I wasn't sure what to change it to.

Reply:
typically I'd prefix the member w/ a 'm_' but apparently the platform is going away from that nomenclature. I'd add the _ to the temporary function param